docs/running_an_mpc_node_in_tdx_external_guide.md

```diff
@@ -199,24 +194,72 @@ echo "  Hash offset:   $HASH_OFFSET"
 echo "  Rootfs hash:   $ROOTFS_HASH"
 
 # Run verification
-sudo veritysetup verify   --data-blocks=$DATA_BLOCKS   --hash-offset=$HASH_OFFSET   --data-block-size=$BLOCK_SIZE   --hash-block-size=$BLOCK_SIZE   "$ROOTFS" "$ROOTFS" "$ROOTFS_HASH"
+sudo veritysetup verify \
+  --data-blocks=$DATA_BLOCKS \
+  --hash-offset=$HASH_OFFSET \
+  --data-block-size=$BLOCK_SIZE \
+  --hash-block-size=$BLOCK_SIZE \
+  "$ROOTFS" "$ROOTFS" "$ROOTFS_HASH"
 
 echo "✅ Verification succeeded"
 ```
 
-**Optional:** Check that your generated OS image corresponds to the expected RTMR values:
+**3. Check actual vs. expected RTMR values:**
+1. Build `dstack-mr` tool (using Docker).  
+2. Calculate the actual RTMRs of the image.  
+3. Compare against expected RTMRs from the contract (shown below).  
+
+For more details, see the [Dstack attestation guide](https://github.com/Dstack-TEE/dstack/blob/master/attestation.md).
 
+Build `dstack-mr` docker image:
 ```bash
-$dstack-mr -cpu 8 -memory 64G -metadata metadata.json
+# Dockerfile
+FROM rust:1.86.0@sha256:300ec56abce8cc9448ddea2172747d048ed902a3090e6b57babb2bf19f754081 AS kms-builder
+ARG DSTACK_REV
+WORKDIR /build
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    git \
+    build-essential \
+    musl-tools \
+    libssl-dev \
+    protobuf-compiler \
+    libprotobuf-dev \
+    clang \
+    libclang-dev
+RUN git clone https://github.com/Dstack-TEE/dstack.git && \
+    cd dstack
+RUN rustup target add x86_64-unknown-linux-musl
+RUN cd dstack && cargo build --release -p dstack-mr-cli --target x86_64-unknown-linux-musl
+
+FROM kvin/kms:latest
+COPY --from=kms-builder /build/dstack/target/x86_64-unknown-linux-musl/release/dstack-mr /usr/local/bin/
+ENTRYPOINT ["dstack-mr"]
+CMD []
+```
+
+Build:
+```bash
+docker build . -t dstack-mr
+```
+
+Run:
+```bash
+docker run --rm \
+  -v "$(pwd)":/dstack-0.5.4 \
+  dstack-mr \
+  measure -c 8 -m 64G /dstack-0.5.4/metadata.json
+```
```

Comment:
So this is just to build the `dstack-mr` tool which we use to verify measurements? Is there any reason why we need to build this as opposed to having phala publish some official binary that could be verified and installed?